### ISSUE TITLE
Panic when using sas_url with azureblob

### ISSUE THREAD

**Issue by tbanach** [ncw/rclone#2969]
#### What is the problem you are having with rclone?

We are unable to use rclone config created using the sas_url option with azureblob. Operations continue to work fine using account name and key. This feature stopped working for us in starting v1.46 (v1.45 works ok).

#### What is your rclone version (output from `rclone version`)

Broken in  v1.46 beta:

```
rclone v1.46.0-016-g1c1a8ef2-beta
- os/arch: linux/amd64
- go version: go1.11.5
```

and v1.46:

```
rclone v1.46
- os/arch: linux/amd64
- go version: go1.11.5
```

, but works in v1.45:

```
rclone v1.45
- os/arch: linux/amd64
- go version: go1.11.2
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

```
Linux 9a5d4f39e4cf 4.9.0-7-amd64 #1 SMP Debian 4.9.110-3+deb9u2 (2018-08-13) x86_64 GNU/Linux
```

####  Which cloud storage system are you using? (eg Google Drive)

Microsoft Azure Blob Storage

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

First define the rclone config using sas_url:

```
rclone config create az azureblob sas_url https://<<<account>>>.blob.core.windows.net/<<<account>>>?sig=<<<secret>>>&se=2019-02-13T16%3A15%3A59Z&sv=2018-03-28&sp=racwdl&sr=c
```

, then try any operation using the config, for example:

```
rclone lsd az:
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```

2019/02/12 16:04:57 DEBUG : rclone: Version "v1.46.0-016-g1c1a8ef2-beta" starting with parameters ["rclone" "-vv" "lsd" "az:"]
2019/02/12 16:04:57 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x85e734]

goroutine 14 [running]:
github.com/ncw/rclone/vendor/github.com/Azure/azure-storage-blob-go/azblob.serviceClient.ListContainersSegment(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-storage-blob-go/azblob/zz_generated_service.go:239 +0x4f4
github.com/ncw/rclone/vendor/github.com/Azure/azure-storage-blob-go/azblob.ServiceURL.ListContainersSegment(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-storage-blob-go/azblob/url_service.go:83 +0x118
github.com/ncw/rclone/backend/azureblob.(*Fs).listContainersToFn.func1(0xc0002f41c0, 0x40bd9f, 0xc000095b80)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/azureblob/azureblob.go:723 +0xe5
github.com/ncw/rclone/lib/pacer.(*Pacer).call(0xc0002f41c0, 0xc000095b80, 0xa, 0x11db820, 0xc0002ce701)
        /home/travis/gopath/src/github.com/ncw/rclone/lib/pacer/pacer.go:405 +0x16a
github.com/ncw/rclone/lib/pacer.(*Pacer).Call(0xc0002f41c0, 0xc000095b80, 0x43d10d, 0xc00006e7d8)
        /home/travis/gopath/src/github.com/ncw/rclone/lib/pacer/pacer.go:428 +0x66
github.com/ncw/rclone/backend/azureblob.(*Fs).listContainersToFn(0xc000136460, 0xc0002865f0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/azureblob/azureblob.go:721 +0x1ca
github.com/ncw/rclone/backend/azureblob.(*Fs).listContainers(0xc000136460, 0x0, 0x0, 0xc0000fe480, 0x0, 0x0, 0xc000047e8c, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/azureblob/azureblob.go:647 +0x91
github.com/ncw/rclone/backend/azureblob.(*Fs).List(0xc000136460, 0x0, 0x0, 0x0, 0x0, 0xc000047f20, 0xc00002e0c0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/azureblob/azureblob.go:669 +0x4d
github.com/ncw/rclone/fs/list.DirSorted(0x14546e0, 0xc000136460, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/fs/list/list.go:23 +0x63
github.com/ncw/rclone/fs/walk.walk.func2(0xc00029fbb0, 0xc0002092c0, 0x12fced8, 0x14546e0, 0xc000136460, 0x0, 0xc00029fb68, 0xc000233480, 0xc00029fbc0, 0xc000231a10, ...)
        /home/travis/gopath/src/github.com/ncw/rclone/fs/walk/walk.go:128 +0x1b7
created by github.com/ncw/rclone/fs/walk.walk
        /home/travis/gopath/src/github.com/ncw/rclone/fs/walk/walk.go:120 +0x252
```

**Comment by ncw** [comment]
A bit of git bisect reveals the problem commit

> commit 5d1d93e163dfa51bc235065eb3220a4f42feca73
> Author: Nick Craig-Wood <nick@craig-wood.com>
> Date:   Fri Nov 30 12:08:22 2018 +0000
> 
>     azureblob: use the rclone HTTP client - fixes #2654
>     
>     This enables --dump headers and --timeout to work properly.

I have figured out the problem, I think - can you give this a go please?

https://beta.rclone.org/branch/v1.46.0-018-g811b8ec7-fix-2969-azureblob-sas-url-beta/ (uploaded in 15-30 mins)

**Comment by tbanach** [comment]
This fixes the problem. Thanks!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.47

Thanks for testing :-)
